src/lib.rs

```diff
@@ -171,7 +171,8 @@ impl<K, V, S> LinkedHashMap<K, V, S> {
         if self.head.is_null() {
             // allocate the guard node if not present
             unsafe {
-                self.head = Box::into_raw(Box::new(mem::uninitialized()));
+                let node_layout = std::alloc::Layout::new::<Node<K, V>>();
+                self.head =  std::alloc::alloc(node_layout) as *mut Node<K, V>;
```

Comment:
Note that `alloc::alloc` is ["expected to be deprecated"](https://doc.rust-lang.org/nightly/std/alloc/fn.alloc.html).

Reply:
yeah but I doubt they'll actually do that since it's strictly worse

Reply:
although Layout also blows so what do I know